### DISCUSSION TITLE
How to change the style of a CustomRichEditBox that has no presence in the XAML

### DISCUSSION THREAD

**Discussion by axtens** [unoplatform/uno#14511] [Q&A] [answered]
A CustomRichEditBox is constructed programmatically and injected into a CustomTabItem. As far as the XAML goes, there's nothing to see:
```
        <NavigationView x:Name="tabControl" PaneDisplayMode="Top" IsBackButtonVisible="Collapsed" IsSettingsVisible="False" RelativePanel.Below="butSelectAll" RelativePanel.AlignLeftWithPanel="True"
                 RelativePanel.AlignRightWithPanel="True"  RelativePanel.Above="outputPanel" SelectionChanged="TabControl_SelectionChanged">
            <NavigationView.MenuItems>
                <local:CustomTabItem x:Name="tab1" Content="Tab 1" Tag="Tab1" IsNewFile="True" Height="30"/>
            </NavigationView.MenuItems>
        </NavigationView>
```
So I have this resource defined
```
        <Style TargetType="RichEditBox" x:Key="CustomRichEditBoxStyle">
            <Setter Property="Background" Value="#F9F8BD"/>
            <!-- Set your desired background color here -->
            <Setter Property="Template">
                <Setter.Value>
                    <ControlTemplate TargetType="RichEditBox">
                        <Border Background="{TemplateBinding Background}" BorderBrush="{TemplateBinding BorderBrush}" BorderThickness="{TemplateBinding BorderThickness}">
                            <ContentPresenter Content="{TemplateBinding Content}" />
                        </Border>
                    </ControlTemplate>
                </Setter.Value>
            </Setter>
        </Style>
```
and I want to apply to the CustomRichEditControl and ordinarily I might expect to the able to do so with ` Style="{StaticResource CustomRichEditBoxStyle}"` but I don't know how to do that programmatically. Presumably, when one instantiates the CustomRichEditBox object one might do something like
```
        public CustomRichEditBox()
        {
            IsSpellCheckEnabled = false;
            IsRTF = true;
            Style = new Style()
            {
              // something useful here?
            }
```
What goes into the `Style`?

**Comment by limeniye** [comment]
Hello, Bruce!

<h3>Answer</h3>

>```Style="{StaticResource CustomRichEditBoxStyle}"``` but I don't know how to do that programmatically

```C#
MyRichEditBox.Style = (Style)Application.Current.Resources["CustomRichEditBoxStyle"];
```

<h3>Comment about Style object</h3>

```C#
Style = new Style()
{
    // something useful here?
}
```

If you open ILSpy, you can see how it works under the hood:
```C#
MarkupHelper.SetElementProperty("ResourceSourceLocations", "CustomRichEditBoxStyle", "file:///R:/ForDelete/UnoApp35/UnoApp35/UnoApp35/UserControl1.xaml#L4:8");
		base.Resources["CustomRichEditBoxStyle"] = new WeakResourceInitializer(this, (object? __ResourceOwner_1) => new Style(typeof(RichEditBox))
		{
			Setters = 
			{
				(SetterBase)new Setter(FrameworkElement.BackgroundProperty, new SolidColorBrush(Color.FromArgb(byte.MaxValue, 249, 248, 189))),
				(SetterBase)new Setter(Control.TemplateProperty, __ResourceOwner_1, (object? __ResourceOwner_2) => new ControlTemplate(__ResourceOwner_2, (object? __owner) => ((I_UserControl1_028a588592bc6aa0b3d399a800901a8b_UnoApp35UserControl1SC0)TypeMappings.CreateInstance<_UserControl1_028a588592bc6aa0b3d399a800901a8b_UnoApp35UserControl1SC0>()).Build(__owner)))
			}
		});
```
But I want to say that doing it via code is a bad idea.
Can you explain why do you need to create the Style object from the code?

<h3>P.S.</h3>

Change  ```Content="{TemplateBinding Content}"``` to  [Content="{TemplateBinding Header}"](https://github.com/unoplatform/uno/blob/5cd6262014b5a0bfe007b20b2955ec232a45acaf/src/Uno.UI.FluentTheme.v1/Resources/Version1/Priority02/RichEditBox_themeresources.xaml#L113), because an error can happens.

**Reply by axtens** [reply]
We asked the contractor to give us the bones of a tabbed text-editor which we are using for an IDE for an in-house programming language.  He opted for the approach that creates an array of tabs hanging off the `<local:CustomTabItem x:Name="tab1" Content="Tab 1" Tag="Tab1" IsNewFile="True" Height="30"/>` line in the XAML.  There are no signs whatsoever of the RichEditBoxes in the XAML. These are all added to the CustomTabItem programmatically. Was it a bad choice? We believed the contractor knew what he was doing. Also we haven't developed enough skills to know better nor do we have the money to invest in those who do.

**Reply by axtens** [reply]
Is your suggestion given in the context of Uno 5? We're currently committed to v4. The error below occurred both before and after I applied your `P.S.`.  
![image](https://github.com/unoplatform/uno/assets/209621/8ff6ec1c-f29e-4493-8f4d-9b6664d586fd)

**Reply by jeromelaban** [reply]
The issue you're seeing is not relate to uno, but rather to WinAppSDK, as you're running the `.Windows` head. Uno 4 vs 5 does not play a role in this issue.

**Comment by axtens** [comment, marked as answer]
Resolved. See https://github.com/unoplatform/uno/discussions/14187
